src/services/codefixes/inferFromUsage.ts

```diff
@@ -274,8 +312,10 @@ namespace ts.codefix {
             return callContexts && declaration.parameters.map((parameter, parameterIndex) => {
                 const types: Type[] = [];
                 const isRest = isRestParameter(parameter);
+                let isOptional = false;
                 for (const callContext of callContexts) {
                     if (callContext.argumentTypes.length <= parameterIndex) {
+                        isOptional = isInJSFile(declaration);

```

Comment:
>isInJSFile [](start = 37, length = 10)

I assume `isInJSFile` is very cheap but I would habitually have either pulled it out of the loop or checked `isOptional ||`.